src/fields/mod.rs

```diff
@@ -198,13 +201,15 @@ pub trait LagrangePolynomialFieldElement: FieldElement {
     }
 }
 
-/// Compute the multiplicative inverse of base, using the provided modulus.
-fn mul_inv_modulus<FE: LagrangePolynomialFieldElement>(base: &FE, modulus: BigUint) -> FE {
-    // We use Euler's theorem to compute inverses. There are more efficient algorithms which we
-    // will adopt in the future.
+/// Compute the multiplicative inverse of base, using the provided order of the field.
+fn mul_inv_field_order<FE: LagrangePolynomialFieldElement>(base: &FE, field_order: BigUint) -> FE {
+    // The multiplicative group of any finite field is a cyclic group with order one less than the
+    // field. See https://www.math.wm.edu/~vinroot/430S13MultFiniteField.pdf for a proof. Therefore,
+    // the order of every element, other than 1, is equal to the group order.
     //
-    // https://en.wikipedia.org/wiki/Modular_multiplicative_inverse#Using_Euler's_theorem
-    base.pow(modulus - (BigUint::one() + BigUint::one()))
+    // We can compute inverses by exponentiating elements to the power of the multiplicative group
+    // order minus one, since (g^(|G|-1)) * g = 1.
```

Comment:
Like, I'm not trying to disprove this paper or assertion, but I'm on a sidequest about Lagrange's theorem and this, because I would have thought that'd be the basis. 

also I should totally have paid more attention in class